src/main/java/org/hl7/davinci/priorauth/SubscriptionEndpoint.java

```diff
@@ -0,0 +1,156 @@
+                SubscriptionChannelType subscriptionType = subscription.getChannel().getType();
+
+                // Check valid subscription type
+                if (subscriptionType == SubscriptionChannelType.RESTHOOK
+                        || subscriptionType == SubscriptionChannelType.WEBSOCKET) {
+                    Subscription processedSubscription = processSubscription(subscription);
+                    if (processedSubscription != null)
+                        formattedData = requestType == RequestType.JSON ? App.getDB().json(processedSubscription)
+                                : App.getDB().xml(processedSubscription);
+                    else {
+                        status = HttpStatus.BAD_REQUEST;
+                        OperationOutcome error = FhirUtils.buildOutcome(IssueSeverity.ERROR, IssueType.INVALID,
+                                PROCESS_FAILED);
+                        formattedData = requestType == RequestType.JSON ? App.getDB().json(error)
+                                : App.getDB().xml(error);
+                    }
+                } else {
+                    // Subscription must be rest-hook or websocket....
+                    status = HttpStatus.BAD_REQUEST;
+                    OperationOutcome error = FhirUtils.buildOutcome(IssueSeverity.ERROR, IssueType.INVALID,
+                            INVALID_CHANNEL_TYPE);
+                    formattedData = requestType == RequestType.JSON ? App.getDB().json(error) : App.getDB().xml(error);
+                }
+            } else {
+                // Subscription is required...
+                status = HttpStatus.BAD_REQUEST;
+                OperationOutcome error = FhirUtils.buildOutcome(IssueSeverity.ERROR, IssueType.INVALID,
+                        REQUIRES_SUBSCRIPTION);
+                formattedData = requestType == RequestType.JSON ? App.getDB().json(error) : App.getDB().xml(error);
+            }
+        } catch (Exception e) {
+            // The subscription failed so spectacularly that we need to
+            // catch an exception and send back an error message...
+            status = HttpStatus.BAD_REQUEST;
+            OperationOutcome error = FhirUtils.buildOutcome(IssueSeverity.FATAL, IssueType.STRUCTURE, e.getMessage());
+            formattedData = requestType == RequestType.JSON ? App.getDB().json(error) : App.getDB().xml(error);
+        }
+        MediaType contentType = requestType == RequestType.JSON ? MediaType.APPLICATION_JSON
+                : MediaType.APPLICATION_XML;
+        return ResponseEntity.status(status).contentType(contentType).body(formattedData);
+    }
+
+    private Subscription processSubscription(Subscription subscription) {
+        // Get the criteria details parsed
+        String claimResponseId = "";
+        String patient = "";
+        String status = "";
+        String criteria = subscription.getCriteria();
+        String regex = "identifier=(.*)&patient.identifier=(.*)&status=(.*)";
```

Comment:
Are the parameters required to appear in the order identifier, patient.identifier, status?   If I send them in a different order is that a legitimate request?

Reply:
Double checked the IG. It says it must be in this form but I missed the note which mentions the order of parameters does not matter. Do you have any suggestions on efficient ways to pull the data of the string if they can appear in any order